### PR TITLE
First draft of offers' API endpoint

### PR DESCRIPTION
**Pull request by magul**
__Story / Bug id:__ https://github.com/CodeForPoznan/volontulo/issues/123

__Reference person:__ @yourgithubname

__Description:__

Description of your changes.

__New imports / dependencies:__

* `package`

__Unit Tests:__

* `apps/volontulo/tests/views/test_auth.py`

__What tests do I need to run to validate this change:__

Description of testing workflow and potential affected places.

### CODE REVIEW

File: apps/volontulo/models.py [hunk 1]
```diff
@@ -226,7 +221,7 @@ def close_offer(self):
 
 
 class UserProfile(models.Model):
-    u"""Model that handles users' profiles."""
+    """Model that handles users' profiles."""
```
Comment (riv2q) [inline apps/volontulo/models.py]: typo

**Comment by tomaszwozniak** [comment]
👍

File: apps/volontulo/tests/models/test_offer.py [hunk 2]
```diff
@@ -11,89 +12,89 @@
...
-            description=u"Dokładny opis oferty",
-            requirements=u"Dokładny opis wymagań",
-            time_commitment=u"333 dni w roku",
-            benefits=u"Wszelkie korzyści z uczestnictwa w wolontariacie",
-            location=u"Polska, Poznań",
-            title=u"Zwięzły tytuł oferty",
-            time_period=u"Od 23.09.2015 do 25.12.2016",
+            description="Dokładny opis oferty",
+            requirements="Dokładny opis wymagań",
+            time_commitment="333 dni w rok",
```
Comment (w1stler) [inline apps/volontulo/tests/models/test_offer.py]: 333 dni w roku ;-)

File: apps/volontulo/tests/models/test_offer.py [hunk 3]
```diff
@@ -103,56 +104,56 @@ def setUp(self):
...
-        u"""Testing offer requirements field"""
-        self.assertEqual(self.offer.requirements, u"Dokładny opis wymagań")
+        """Testing offer requirements field"""
+        self.assertEqual(self.offer.requirements, "Dokładny opis wymagań")
 
     def test__offer_time_commit_field(self):
-        u"""Testing offer time commitment field"""
-        self.assertEqual(self.offer.time_commitment, u"333 dni w roku")
+        """Testing offer time commitment field"""
+        self.assertEqual(self.offer.time_commitment, "333 dni w rok")
```
Comment (w1stler) [inline apps/volontulo/tests/models/test_offer.py]: ditto

File: apps/volontulo/tests/views/api/offers/test_list.py [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    def setUp(self):
+        """Set up each test."""
+        super(TestRegularUserOffersListAPIView, self).setUp()
+        self.client.login(
+            username='volunteer@example.com',
+            password='123volunteer'
+        )
+
+    def test_offer_list_length(self):
+        """Test offers' list length for regular user.
```
Comment (w1stler) [inline apps/volontulo/tests/views/api/offers/test_list.py]: Redundant '

File: apps/volontulo/views/api.py [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+
+from apps.volontulo import models
+from apps.volontulo import permissions
+from apps.volontulo import serializers
+from apps.volontulo.views import logged_as_admin
+
+
+class OfferViewSet(viewsets.ModelViewSet):
+
+    """REST API offers' viewset."""
```
Comment (w1stler) [inline apps/volontulo/views/api.py]: Redundant '

File: apps/volontulo/views/api.py [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    def get_queryset(self):
+        """Queryset depends on user role."""
+        if logged_as_admin(self.request):
+            return models.Offer.objects.get_for_administrator()
+        return models.Offer.objects.get_weightened()
+
+
+class OrganizationViewSet(viewsets.ModelViewSet):
+
+    """REST API organizations' viewset."""
```
Comment (w1stler) [inline apps/volontulo/views/api.py]: Ditto

**Comment by riv2q** [comment]
LF

**Pull request merged**